fvm/transactionVerifier.go

```diff
@@ -13,24 +13,31 @@ import (
 	"github.com/onflow/flow-go/module/trace"
 )
 
-type TransactionSignatureVerifier struct {
+// TransactionVerifier verifies the content of the transaction by
+// checking accounts (authorizers, payer, proposer) are not frozen
+// checking there is no double signature
+// all signatures are valid
+// all accounts provides enoguh weights
+//
+// if KeyWeightThreshold is set to a negative number, signature verification is skipped
+type TransactionVerifier struct {
```

Comment:
Whats the benefit of merging all the checks into one processor?

I was thinking that the frozen checker could actually be run for all transactions in a block(chunk?) in parallel before any transaction invocation. This would only change the behaviour of the freeze so that it is valid only from the next block on, which I think would be acceptable.

Reply:
I think checking frozen accounts should happen at the time of  the register reads, for example, we do have extra checks when a contract is going to load from an account that is not the authorizer of the tx but should still block that call, etc ...